src/zkp/pifac.rs

```diff
@@ -467,23 +482,37 @@ mod tests {
         with_random_no_small_factors_proof(&mut rng, correct_factors)
     }
 
-    /// TODO: This test does not currently work. It should be addressed as part
-    /// of issue #113.
-    // #[test]
-    #[allow(dead_code)]
-    fn test_modulus_cannot_have_small_factors() -> Result<()> {
+    #[test]
+    fn test_modulus_cannot_have_large_factors() -> Result<()> {
```

Comment:
suggestion (style preference): I think this would be clearer if we distinguished less between `p` and `q` in the variable naming. The idea is that we have a big factor (product of two large primes) and a small factor (2) and the modulus is a product of those two things. We don't have to separately rename them depending in which one we pass first in the secret.

If I were writing this, I would make a `small_factor` and `big_factor`, multiply them to get the `modulus`, and then reuse those three fields for both proofs, rather than cloning and re-calculating stuff.